libkpod/rename.go

```diff
@@ -0,0 +1,108 @@
+package libkpod
+
+import (
+	"encoding/json"
+	"path/filepath"
+
+	"github.com/docker/docker/pkg/ioutils"
+	"github.com/kubernetes-incubator/cri-o/oci"
+	"github.com/opencontainers/runtime-tools/generate"
+)
+
+const (
+	configFile        = "config.json"
+	nameAnnotation    = "io.kubernetes.cri-o.Name"
+	metadatAnnotation = "io.kubernetes.cri-o.Metadata"
+)
+
+// ContainerRename renames the given container
+func (c *ContainerServer) ContainerRename(container, name string) error {
+	ctr, err := c.LookupContainer(container)
+	if err != nil {
+		return err
+	}
+
+	oldName := ctr.Name()
+	_, err = c.ReserveContainerName(ctr.ID(), name)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err != nil {
+			c.ReleaseContainerName(oldName)
+		} else {
+			c.ReleaseContainerName(oldName)
+		}
+	}()
+
+	// Update state.json
+	if err = c.updateStateName(ctr, name); err != nil {
+		return err
+	}
+
+	// Update config.json
+	configRuntimePath := filepath.Join(ctr.BundlePath(), configFile)
+	if err = updateConfigName(configRuntimePath, name); err != nil {
+		return err
+	}
+	configStoragePath := filepath.Join(ctr.Dir(), configFile)
+	if err = updateConfigName(configStoragePath, name); err != nil {
+		return err
+	}
+
+	// Update containers.json
+	return nil
+}
+
+func (c *ContainerServer) updateContainerStoreName(id, name string) error {
+	cstore, err := c.store.ContainerStore()
+	if err != nil {
+		return err
+	}
+
+	return cstore.SetNames(id, []string{name})
```

Comment:
Use `c.store.SetNames` here.  Accessing the `ContainerStore` directly bypasses synchronization and locking that `c.store` provides.